### DISCUSSION TITLE
Disable leader key inside tmux

### DISCUSSION THREAD

**Discussion by OkanEsen** [wezterm/wezterm#2441] [Q&A] [answered]
Sometimes I have to use tmux inside wezterm, when I'm connected to a remote machine. When I'm inside a tmux session, the leader key on both wezterm and tmux are identical to make my life easier and not mess up my muscle memory.

I wanted to implement some kind of behavior, where I trigger a [custom event ](https://wezfurlong.org/wezterm/config/keys.html?search=leader#configuring-key-assignments), check if I'm inside of tmux and disable the leader key in wezterm, as soon as I'm inside a tmux session but I think that this is not possible, correct?

In short:

Inside tmux -> disable wezterm leader
Otherwise -> use wezterm leader

**Comment by wez** [comment, marked as answer]
I'd suggest defining an update-status event handler (https://wezfurlong.org/wezterm/config/lua/window-events/update-status.html) and in the handler, use https://wezfurlong.org/wezterm/config/lua/pane/get_foreground_process_name.html to see if tmux is running, then use https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html to adjust the leader key configuration.

If you run tmux on a remote system then wezterm won't have a direct way to know if tmux is running because it won't be able to inspect the processes there. You might consider configuring tmux to emit a user var (eg: `IN_TMUX=true`) that you can test in the update-status event handler instead: https://wezfurlong.org/wezterm/config/lua/pane/get_user_vars.html briefly shows how to use user vars.  Another option is to look at the window title instead: https://wezfurlong.org/wezterm/config/lua/pane/get_title.html

**Reply by OkanEsen** [reply]
Thank you so much for your thorough answer.

I tried to use the `update-status` event and was unfortunately not able to get the information I need from the pane object. I think this is obviously expected though, because I forgot to say, that I'm using the wezterm multiplexer too and  connect to it by default:

```lua
...
  unix_domains = {{name = "default"}},
  default_gui_startup_args = {"connect", "default"},
...
```

Based on the readme, I won't be able to get the information I need:

> This information is only available for local panes. Multiplexer panes do not report this information. Similarly, if you are using eg: ssh to connect to a remote host, you won't be able to access the name of the remote process that is running.
> Source: https://wezfurlong.org/wezterm/config/lua/pane/get_foreground_process_name.html

Your second paragraph already addresses that so using `get_user_vars` should get me want I want.

Thanks!
